### DISCUSSION TITLE
Create a client for Node.js

### DISCUSSION THREAD

**Discussion by handro1** [OddDotNet/OddDotNet#40] [Ideas]
I didn't see an issue for creating a client for Node.js. I think you can using something like this https://github.com/grpc/grpc-node. I'm learning about Node.js and could possibly help if you like this idea haven't got to it yet.

**Comment by DoubleTK** [comment]
Yeah that sounds great. Obviously I'd like to create clients and builders for all the mainstream languages, but time is a limiting factor so I chose C#/Java/Go to start with. Generating the client shouldn't be too difficult. The builder is a bit more complicated, but I'm pretty sure we can just re-use the pattern I'm following with C# across the other languages. 

I can get a new repo created called `OddDotNode`. I'm following the repo naming convention of `OddDot{LanguageOrFramework}` for where the client code lives. I can do that for this one too, or you can do it under your own account and give it the same name or a different one, it just won't be under the `OddDotNet` org in that scenario. 

You'll need to pull in the `OddDotProto` repo as a git submodule most likely. Let me know if you want help with that.

**Reply by handro1** [reply]
Sounds good. I should be able to work on this next week. Maybe you can make the repo so that it's under the same `OddDotNet` org and I can take it from there?

**Reply by DoubleTK** [reply]
Sounds like a plan 👍
